packages/sheets-ui/src/controllers/permission/sheet-permission-interceptor-canvas-render.controller.ts

```diff
@@ -72,26 +72,16 @@ export class SheetPermissionInterceptorCanvasRenderController extends RxDisposab
                         return true;
                     }
 
-                    const protectionLapRange = this._rangeProtectionRuleModel.getSubunitRuleList(unitId, subUnitId).reduce((p, c) => {
-                        return [...p, ...c.ranges];
-                    }, [] as IRange[]).filter((range) => {
-                        return Rectangle.intersects(range, selectionRange);
-                    });
-
-                    const haveNotPermission = protectionLapRange.some((range) => {
-                        const { startRow, startColumn, endRow, endColumn } = range;
-                        for (let row = startRow; row <= endRow; row++) {
-                            for (let col = startColumn; col <= endColumn; col++) {
-                                const permission = (worksheet.getCell(row, col) as (ICellDataForSheetInterceptor & { selectionProtection: ICellPermission[] }))?.selectionProtection?.[0];
-                                if (permission?.[UnitAction.Edit] === false) {
-                                    return true;
-                                }
+                    const permissionIds: string[] = [];
```

Comment:
Could this be done in a single loop? 

Supposing that the first protection rule forbids the command, the current code would still loop through all rules and ranges. But if you do it in a single loop, you can skip all other rules.